Assets/Scripts/Enemy/StateView/PetrolState.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using BTManager;
+
+namespace Enemy
+{
+    public class PetrolState : EnemyBaseStateView
+    {
+        [SerializeField] private EnemyView enemyView;
+
+        private List<Vector3> wayPointList;
+
+        private int currentWayPointIndex;
+
+        [SerializeField]
+        private float moveLimit;
+
+        private int replayIndexCOunt = 0;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+
+
+            Debug.Log(this.name + " is in PetrolState");
+            wayPointList = new List<Vector3>();
+
+            foreach (var wayPoint in FindObjectsOfType<WayPoints>())
+            {
+                wayPointList.Add(wayPoint.GetComponent<Transform>().position);
+            }
+
+            if (GameManager.Instance.currentState.gameStateType == StateMachine.GameStateType.Game)
+            {
+                SelectWayPoint();
+            }
+            else if (GameManager.Instance.currentState.gameStateType == StateMachine.GameStateType.Replay)
+            {
+                currentWayPointIndex = EnemyManager.Instance.EnemyDatas[enemyView.enemyIndex].wayPoints[replayIndexCOunt];
```

Comment:
the waypoint data needs to be passed into the view from the controller. view should not be going out to services to get data